extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddNodePortDecorator.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.dekorate.deps.kubernetes.api.model.ObjectMeta;
+import io.dekorate.deps.kubernetes.api.model.ServiceSpecFluent;
+import io.dekorate.kubernetes.decorator.Decorator;
+import io.dekorate.kubernetes.decorator.NamedResourceDecorator;
+
+public class AddNodePortDecorator extends NamedResourceDecorator<ServiceSpecFluent> {
+
+    private static final int MIN_VALUE = 30000;
+    private static final int MAX_VALUE = 31999;
+
+    private final int nodePort;
+
+    public AddNodePortDecorator(String name, int nodePort) {
+        super(name);
+        if (nodePort < MIN_VALUE || nodePort > MAX_VALUE) {
```

Comment:
This https://kubernetes.io/docs/concepts/services-networking/service/#nodeport says that the default range is 30000-32767, and apparently it's configurable.

Reply:
> I am a bit skeptical about using a fixed node port.
> 
> What will happen if for any reason the port is already taken?
> What will happen if I am using the `kubernetes` extension in multiple applications or modules?
> 
> So, I would like to do some small research for alternatives.
> 
> If no other alternative can be found, then I would add the following behavior:
> 
> 1. Feature flag to turn this on and off (default value is debatable).

Why would you need a feature flag? Isn't the use of `NodePort` along with `1` replica enough?

> 2. I would hash the service name and namespace and get the default node port (to prevent conflicts if no node port has been explicitly specified).

Yes, this is pretty much what I suggested above.

> 3. Deployer should display the node port to the user.

Good idea

Reply:
Ah, nice find! I need to remove these checks then.